### DISCUSSION TITLE
[QUESTION]:ilogtail的压测结果和阿里云官网文档提供的数据差距很大

### DISCUSSION THREAD

**Discussion by xiongjunkun** [alibaba/ilogtail#1143] [Help] [answered]
目前正在调研各种日志采集客户端的性能，所以针对ilogtail进行了一系列压测，发现压测结果和阿里云官网提供的数据差距很大，有人知道原因吗？

我的压测步骤：
1、提前在节点中构造20G的日志文件：5个日志文件，每个日志文件的大小是4G，并且日志内容是随机的。
2、在K8s中部署ilogtail，并且在ilogtail容器的环境变量中配置ilogtail的资源规格：
        - name: cpu_usage_limit
          value: "1"
        - name: mem_usage_limit
          value: "1024"
        - name: max_bytes_per_sec
          value: "209715200"
        - name: process_thread_count
          value: "8"
        - name: send_request_concurrency
          value: "20"

最后压测的结果，极简模式的采集速度是 38.94 MB/s 。
然后我将cpu_usage_limit和mem_usage_limit调整为2c2g，发现极简模式的采集速度还是38M左右，远远低于阿里元官网说明的100M/s

**Comment by messixukejia** [comment]
https://github.com/alibaba/ilogtail/discussions/809
https://mp.weixin.qq.com/s/8mCVk3gvXPOijTlcRjUR_w

可以参考下

**Comment by xiongjunkun** [comment]
> #809 https://mp.weixin.qq.com/s/8mCVk3gvXPOijTlcRjUR_w
> 
> 可以参考下

我的投递目标是SLS的logstroe，并且我的数据是随机的，这篇文档的测试数据应该都是相同的。

**Comment by linrunqi08** [comment]
@xiongjunkun  1. 测试的网络环境是怎么样的呢，到sls使用的是公网还是阿里云内网？ 2. logstore的shard数量是多少？单shard写入能力是有上限的，具体可以看下sls 官方文档

**Comment by xiongjunkun** [comment]
> 

1、走的公网，我自己机器上面的k8s。
2、shard数量是10，并且我开启自动分裂了，自动分裂的上限是64。

**Comment by linrunqi08** [comment, marked as answer]
比较大的可能还是网络瓶颈。
1. shard建议设置到32，自动分裂条件比较苛刻
2. send_request_concurrency 可以设置到80看下
3. 使用同地域ecs走内网intranet进行测试

**Comment by messixukejia** [comment]
ilogtail默认采集动态数据，所以“提前在节点中构造20G的日志文件”的方法是需要调整下。

**Comment by messixukejia** [comment]
可以使用https://github.com/iLogtail/mock-log 构造动态数据
